Bugfix: bunch of bugfixes

ImJustKisik opened the pull request:
fixes: #161
fixes: #162
fixes: #151
fixes: #159 
fixes: #203
fixes: #145 
fixes: https://github.com/Baystation12/Baystation12/issues/32904
 

<details>
<summary>Чейнджлог</summary>

```yml
🆑
bugfix: ППТ больше не ощущают голода
bugfix: Заложник теперь защищает только спереди
bugfix: Добавлена задержка между ударами мобом по стеклу
bugfix: У Flash shell - исправлен спрайт 
bugfix: Больше нельзя защитить себя собой 
tweak: Локализация верба модпаков
/🆑
```

</details>


- [X] Pull Request полностью завершен, мне не нужна помощь чтобы его закончить.
- [X] Я внимательно прочитал все свои изменения и багов в них не нашел.
- [X] Я запускал сервер со своими изменениями локально и все протестировал.

ImJustKisik committed:
fix of FPB alert

ImJustKisik committed:
grab fix

ImJustKisik committed:
icon fix

ImJustKisik committed:
modpack localisation

ImJustKisik committed:
dir. grab

ImJustKisik committed:
fix self red grab (need tests)

SuhEugene commented on code/modules/mob/living/carbon/human/life.dm:
```diff
@@ -783,36 +783,36 @@
+			switch(hydration)
+				if(395 to INFINITY)
+					throw_alert("hydration", over_hydrated)
+				if(390 to 395)
+					throw_alert("hydration", hydrated)
+				if(250 to 390)
+					clear_alert("hydration")
```
Нет. Не надо путать игроков. Как оно было раньше (до алёртов), так оно и должно быть сейчас

```suggestion
				if(390 to 395)
					clear_alert("hydration")
				if(250 to 390)
					throw_alert("hydration", hydrated)
```

SuhEugene commented on code/modules/mob/living/carbon/human/life.dm:
```diff
@@ -783,36 +783,36 @@
+			switch(nutrition)
+				if(395 to INFINITY)
+					throw_alert("nutrition", fat_alert)
+				if(390 to 395)
+					throw_alert("nutrition", well_fed)
+				if(250 to 390)
+					clear_alert("nutrition")
```
Нет. Не надо путать игроков. Как оно было раньше (до алёртов), так оно и должно быть сейчас

```suggestion
				if(390 to 395)
					clear_alert("nutrition")
				if(250 to 390)
					throw_alert("nutrition", well_fed)
```

ImJustKisik replied:
Всё так как и должно быть. Это сделано, чтобы большую часть времени на экране было пусто. Смысл алертов в том, чтобы оповещать об изменениях, а не быть постоянным индикатором

SuhEugene replied:
В таком случае велл фед однозначно должен быть синего цвета. Это **очень** сильно путает.

Я понял о чём ты ведёшь речь, я только сейчас заметил что это всего от 390 до 395.
Но всё-равно неприятная такая дичь с зелёным индикатором...

ImJustKisik replied:
Поправил иконки

SuhEugene commented on code/modules/projectiles/ammunition/bullets.dm:
```diff
@@ -127,8 +127,8 @@
 	desc = "A chemical shell used to signal distress or provide illumination."
-	icon_state = "fshell"
-	spent_icon = "fshell-spent"
+	icon_state = "pshell"
+	spent_icon = "pshell-spent"
```
```suggestion
	// icon_state = "fshell" // BAY
	// spent_icon = "fshell-spent" // BAY
	icon_state = "pshell" // SIERRA
	spent_icon = "pshell-spent" // SIERRA
```

SuhEugene commented on code/modules/projectiles/projectile.dm:
Это бы на оффы закоммитить..........

SuhEugene commented on code/controllers/subsystems/initialization/modpacks.dm:
```diff
@@ -44,7 +44,7 @@ SUBSYSTEM_DEF(modpacks)
 
 	if(length(SSmodpacks.loaded_modpacks))
-		. = "<hr><br><center><b><font size = 3>Modpacks List</font></b></center><br><hr><br>"
+		. = "<hr><br><center><b><font size = 3>Список Модификаций</font></b></center><br><hr><br>"
```
Би рашн, нот американ
```suggestion
		. = "<hr><br><center><b><font size = 3>Список модификаций</font></b></center><br><hr><br>"
```

SuhEugene commented on code/controllers/subsystems/initialization/modpacks.dm:
```diff
@@ -55,13 +55,13 @@ SUBSYSTEM_DEF(modpacks)
 				. += "</div><br>"
 
-		var/datum/browser/popup = new(mob, "modpacks_list", "Modpacks List", 480, 580)
+		var/datum/browser/popup = new(mob, "modpacks_list", "Список Модификаций", 480, 580)
```
```suggestion
		var/datum/browser/popup = new(mob, "modpacks_list", "Список модификаций", 480, 580)
```

SuhEugene commented on code/game/objects/structures/window.dm:
```diff
@@ -449,6 +449,8 @@
 			grab.affecting.apply_damage(20, DAMAGE_BRUTE, def_zone, used_weapon = src)
 			hit(50, grab.assailant, grab.affecting)
+
+		qdel(grab)
```
```suggestion
		qdel(grab) // SIERRA
```

ImJustKisik committed:
i hate taste of code review in my bugfixes

Update code/game/objects/structures/window.dm

Update code/controllers/subsystems/initialization/modpacks.dm

Co-Authored-By: SuhEugene <32931701+SuhEugene@users.noreply.github.com>

ImJustKisik committed:
СКОЛЬКО ЮЖИКА НЕ КОРМИ ОН ВСЕ В КОД СМОТРИТ

SuhEugene committed:
Blue fed alert color + fix of blue hydration
